--- server/collab-server.js	2015-06-18 15:29:30.028239130 +0500
+++ server/collab-server.js	2015-06-18 12:45:02.561540833 +0500
@@ -1142,6 +1142,8 @@ function handleConnect(userIds, client)
             var usersMap = {};
             users.forEach(function (user) {
                 var uid = user.uid;
+                // Skipping guests
+                if (uid >= 1000000) return;
                 var onlineUserClients = onlineUsers[uid] || [];
                 var onlineState;
                 if (idleUsers[uid])
@@ -1415,6 +1417,9 @@ function handleChatMessage(userIds, clie
     var text = data.text;
     var userId = userIds.userId;
 
+    // Block messages from guests
+    if (userId >= 1000000) return;
+    
     // Save the chat message and broadcast it
     Store.saveChatMessage(text, userId, function (err, message) {
       if (err)
@@ -1464,6 +1469,7 @@ function handleCursorUpdate(userIds, cli
  * @param {String} docId   - the document id or path
  */
 function broadcast(message, sender, docId) {
+    if (sender && sender.userIds.userId >= 1000000) return;
     var toClientIds = docId ? documents[docId] : clients;
     var audienceNum = 0;
     for (var clientId in toClientIds) {
